### PR TITLE
Processing a control message causes the outbox to throw a null reference exception

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

File: src/SharedAcceptanceTests.All/When_using_outbox_control_message.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                Context testContext;
+
+                public ControlMessageBehavior(Context testContext)
+                {
+                    this.testContext = testContext;
+                }
+
+                public override async Task Invoke(IIncomingPhysicalMessageContext context, Func<Task> next)
+                {
+                    await next();
```
Comment (andreasohlund) [inline src/SharedAcceptanceTests.All/When_using_outbox_control_message.cs:79]: This triggers the null ref

File: src/PhysicalOutbox.StorageTable.AcceptanceTests/ConfigureAzureTablePersistence.cs [hunk 2]
```diff
@@ -32,8 +32,10 @@ Task IConfigureEndpointTestExecution.Configure(string endpointName, EndpointConf
             }
 
             // This populates the partition key at the physical stage to test the conventional outbox use-case
-            configuration.Pipeline.Register(typeof(PartitionKeyProviderBehavior), "Populates the partition key");
-
+            if (endpointName != Conventions.EndpointNamingConvention(typeof(When_using_outbox_control_message.Endpoint)))
```
Comment (andreasohlund) [inline src/PhysicalOutbox.StorageTable.AcceptanceTests/ConfigureAzureTablePersistence.cs:35]: Not proud of this but similar to line 29 above

Comment (danielmarbach) [reply src/PhysicalOutbox.StorageTable.AcceptanceTests/ConfigureAzureTablePersistence.cs:35]: In Cosmos this worked really well

https://github.com/Particular/NServiceBus.Persistence.CosmosDB/blob/master/src/SharedAcceptanceTests.RequirePartitionKey/RunSettingsExtensionsRequirePartitionKey.cs#L7-L8

https://github.com/Particular/NServiceBus.Persistence.CosmosDB/blob/8c8a04078d00415ad83ddc4b7d1465fff3690c47/src/NServiceBus.Persistence.CosmosDB.PhysicalOutbox.AcceptanceTests/ConfigureEndpointCosmosDBPersistence.cs#L26-L33

https://github.com/Particular/NServiceBus.Persistence.CosmosDB/blob/8c8a04078d00415ad83ddc4b7d1465fff3690c47/src/NServiceBus.Persistence.CosmosDB.LogicalOutbox.AcceptanceTests/When_custom_partition_key_extractor_from_message_registered_via_di.cs#L19-L20

Comment (andreasohlund) [reply src/PhysicalOutbox.StorageTable.AcceptanceTests/ConfigureAzureTablePersistence.cs]: Thanks, fixed now

**Comment by andreasohlund** [comment]
@danielmarbach what do U think of the bugfix?

**Comment by andreasohlund** [comment]
@danielmarbach added the tests and checks that we agreed on

File: src/PhysicalOutbox.StorageTable.AcceptanceTests/When_no_partition_key_is_configured.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+{
+    using System.Linq;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NServiceBus.AcceptanceTesting.Support;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class When_no_partition_key_is_configured : NServiceBusAcceptanceTest
```
Comment (danielmarbach) [inline src/PhysicalOutbox.StorageTable.AcceptanceTests/When_no_partition_key_is_configured.cs:11]: Both the test would also be true for the logical outbox part not just the physical right? So if we did move them into the appropriate shared folder and added a similar table client holder check right before https://github.com/Particular/NServiceBus.Persistence.AzureTable/blob/master/src/NServiceBus.Persistence.AzureTable/Outbox/LogicalOutboxBehavior.cs#L57-L58 those acceptance tests would cover both phases properly right?

Comment (andreasohlund) [reply src/PhysicalOutbox.StorageTable.AcceptanceTests/When_no_partition_key_is_configured.cs:11]: Yea, I’ll give it a try

Comment (andreasohlund) [reply src/PhysicalOutbox.StorageTable.AcceptanceTests/When_no_partition_key_is_configured.cs]: Moved both now, please take a look

**Review by danielmarbach — APPROVED** [review]

**Pull request merged**